server/src/main/java/org/opensearch/search/internal/ContextIndexSearcher.java

```diff
@@ -271,6 +293,10 @@ protected void search(List<LeafReaderContext> leaves, Weight weight, Collector c
         // This is actually beneficial for search queries to start search on latest segments first for time series workload.
         // That can slow down ASC order queries on timestamp workload. So to avoid that slowdown, we will reverse leaf
         // reader order here.
+        if (searchContext.shouldUseConcurrentSearch()) {
+            fromConcurrentPath = true;
+        }
```

Comment:
1. I think the comment above needs to stay with the code below
2. Do we actually need this `fromConcurrentPath` local variable instead of just doing `shouldUseConcurrentSearch()` each time? It looks like this is only used in `ContextIndexSearcher` which should have `searchContext` throughout.

Reply:
1) Updated
2) `shouldUseConcurrentSearch()` only give us if this search query is concurrent search enabled or not. However, in the concurrent search enabled case, query rewrite could happen on "before" (i.e. non-concurrent path) and "during" the concurrent path. So we need a flag (i.e. `fromConcurrentPath`) to separate the non-concurrent path and concurrent path cases in `ContextIndexSearcher search()`, where is the concurrent path actual happened place.